src/adv.py

```diff
@@ -33,19 +31,72 @@
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
 
-#
-# Main
-#
-
-# Make a new player object that is currently in the 'outside' room.
-
-# Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
-#
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
-#
-# If the user enters "q", quit the game.
+
+def startAdventure():
+    end = False
+    start = True
+    while end == False:
+        if start == True:
+            newPlayer = Player(room["outside"])
+            start = False
+        print("\n")
+        print(newPlayer.room)
+        print("\n")
+        if (len(newPlayer.room.items) == 0):
+            print("There are no items in the room")
+        else:
+            print("The items in the room are: ")
+            for itemInstance in newPlayer.room.items:
+                print(f"  {itemInstance.name} ({itemInstance.description})\n")
+
+        userInput = input(
+            "Which way will you go? [n, s, e, w, or q to quit (or i for inventory)]:\n\n\n\n")
+
+        try:
+            if userInput == "n":
+                newPlayer.room = newPlayer.room.n_to
+            elif userInput == "s":
+                newPlayer.room = newPlayer.room.s_to
+            elif userInput == "e":
+                newPlayer.room = newPlayer.room.e_to
+            elif userInput == "w":
+                newPlayer.room = newPlayer.room.w_to
+            elif userInput == "i":
+                print("Your inventory:\n")
+                for invItem in newPlayer.inventory:
+                    print(f"{invItem.name}\n")
+
+                roomItemNames = [x.name for x in newPlayer.room.items]
+                inventoryItemNames = [x.name for x in newPlayer.inventory]
+
+                userInputItemAdd = input(
+                    "Your options: \n'Get <itemName>' \n 'Take <itemName> \n 'Drop <itemName> \n 'Remove <itemName>? \n")
+
+                method = userInputItemAdd.split(" ")[0]
+                itemToChange = userInputItemAdd.split(" ")[1]
+
+                if method == "get" or method == "take":
+                    if itemToChange in roomItemNames:
+                        itemIndex = roomItemNames.index(itemToChange)
+                        newPlayer.addItem(newPlayer.room.items[itemIndex])
+                    else:
```

Comment:
this section could definitely be refactored for readability
- look for anything that could be broken out into a function
- avoid using multiple list methods to assign variable when one will do